coriolis/tests/osmorphing/test_manager.py

```diff
@@ -0,0 +1,351 @@
+# Copyright 2024 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import logging
+from unittest import mock
+
+from coriolis import exception
+from coriolis.osmorphing import base as base_osmorphing
+from coriolis.osmorphing import manager
+from coriolis.tests import test_base
+
+
+class ManagerTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis manager module."""
+
+    def setUp(self):
+        super(ManagerTestCase, self).setUp()
+        self.osmorphing_info = {
+            'os_type': 'linux',
+            'ignore_devices': []
+        }
+        self.provider = mock.MagicMock()
+        self.event_handler = mock.MagicMock()
+        self.os_mount_tools = mock.MagicMock()
+        self.event_manager = mock.MagicMock()
+        self.destination_provider = mock.MagicMock()
+        self.worker_connection = mock.MagicMock()
+        self.detected_os_info = mock.MagicMock()
+
+    @mock.patch('coriolis.osmorphing.manager.CONF')
```

Comment:
Please avoid mocking `coriolis.osmorphing.manager.CONF`.

In the `setUp` method, you can do the following to set actual values for the conf options:
```python
manager.CONF.proxy.url = "http://127.0.0.1:8080"
manager.CONF.proxy.username = "admin"
manager.CONF.proxy.password = "Random-Password-123!"
manager.CONF.proxy.no_proxy = ["cloudbase.it"]
```

and the `test_get_proxy_settings` unit test body looks like this:
```python
def test_get_proxy_settings(self):
    expected_result = {
        "url": "http://127.0.0.1:8080",
        "username": "admin",
        "password": "Random-Password-123!",
        "no_proxy": ["cloudbase.it"],
    }

    result = manager._get_proxy_settings()
    self.assertEqual(result, expected_result)

```